tensorflow/core/util/gpu_device_functions.h

```diff
@@ -83,39 +93,52 @@ class CudaGridRange {
   T end_;
 };
 
+#ifndef TENSORFLOW_USE_ROCM
+template <typename... T>
+using CudaGridRange = GpuGridRange<T...>;
+#endif
 }  // namespace detail
 
 // Helper to visit indices in the range 0 <= i < count, using the x-coordinate
 // of the global thread index. That is, each index i is visited by all threads
 // with the same x-coordinate.
-// Usage: for(int i : CudaGridRangeX(count)) { visit(i); }
+// Usage: for(int i : GpuGridRangeX(count)) { visit(i); }
 template <typename T>
-__device__ detail::CudaGridRange<T> CudaGridRangeX(T count) {
-  return detail::CudaGridRange<T>(blockIdx.x * blockDim.x + threadIdx.x,
+__device__ detail::GpuGridRange<T> GpuGridRangeX(T count) {
+  return detail::GpuGridRange<T>(blockIdx.x * blockDim.x + threadIdx.x,
                                   gridDim.x * blockDim.x, count);
 }
+CREATE_CUDA_DEVICE_FUNCTION_ALIAS(GpuGridRangeX, CudaGridRangeX);
 
 // Helper to visit indices in the range 0 <= i < count using the y-coordinate.
-// Usage: for(int i : CudaGridRangeY(count)) { visit(i); }
+// Usage: for(int i : GpuGridRangeY(count)) { visit(i); }
 template <typename T>
-__device__ detail::CudaGridRange<T> CudaGridRangeY(T count) {
-  return detail::CudaGridRange<T>(blockIdx.y * blockDim.y + threadIdx.y,
+__device__ detail::GpuGridRange<T> GpuGridRangeY(T count) {
+  return detail::GpuGridRange<T>(blockIdx.y * blockDim.y + threadIdx.y,
                                   gridDim.y * blockDim.y, count);
 }
+CREATE_CUDA_DEVICE_FUNCTION_ALIAS(GpuGridRangeY, CudaGridRangeY);
 
 // Helper to visit indices in the range 0 <= i < count using the z-coordinate.
-// Usage: for(int i : CudaGridRangeZ(count)) { visit(i); }
+// Usage: for(int i : GpuGridRangeZ(count)) { visit(i); }
 template <typename T>
-__device__ detail::CudaGridRange<T> CudaGridRangeZ(T count) {
-  return detail::CudaGridRange<T>(blockIdx.z * blockDim.z + threadIdx.z,
+__device__ detail::GpuGridRange<T> GpuGridRangeZ(T count) {
+  return detail::GpuGridRange<T>(blockIdx.z * blockDim.z + threadIdx.z,
                                   gridDim.z * blockDim.z, count);
 }
+CREATE_CUDA_DEVICE_FUNCTION_ALIAS(GpuGridRangeZ, CudaGridRangeZ);
 
+#if GOOGLE_CUDA
 // Mask for all 32 threads in a warp.
-const unsigned kCudaWarpAll = 0xffffffff;
+__device__ const unsigned kCudaWarpAll = 0xffffffff;
+#elif TENSORFLOW_USE_ROCM
+// ROCM TODO add ROCM implementation
+// Mask for all 64 threads in a wavefront.
+__device__ const unsigned kCudaWarpAll = 0xffffffff;
```

Comment:
The two are the same, could you just leave the comment but remove the #if/else?

Reply:
Make sense, will do.

Reply:
Taking a step back, I feel like keeping it and get the ROCm side of the `kCudaWarpAll` to be renamed to `kGpuWarpAll`, just to be consistent to the aim of creating alias. Since the aim was to make the `Gpu` in variable naming so that it works for both ROCm and Cuda. 

This is so that when dependent kernels that uses `kCudaWarpAll`, as well as line 133 will be renamed to kGpuWarpAll as well. How does that sound?

Reply:
We have been providing both symbols during a rename. If you just rename the ROCm side, we won't be able to change usages one by one.

Reply:
Sounds good. Will remove #if/else and make both kCudaWarpAll and kGpuWarpAll available to be consistent with the reset of PR.